src/Covenant/Internal/Unification.hs

```diff
@@ -236,10 +302,38 @@ unify expected actual =
           then noSubUnify
           else unificationError
       _ -> unificationError
+
+    expectDatatype :: TyName -> Vector (ValT Renamed) -> UnifyM (Map (Index "tyvar") (ValT Renamed))
```

Comment:
Might be worth adding a 'spell-out' comment similar to the ones with the other `expect*` helpers for consistency. While it's pretty clear what's going on here at a glance, `concretify` especially is unique, and might warrant some (non-Haddock) explaining, if only for our future selves.